Refactor: separate CI-specific weight validation into dedicated module

alisonshao opened the pull request:
## Summary

- Rename `weight_validation.py` to `ci_weight_validation.py` to clearly indicate its CI-specific purpose
- Add wrapper functions (`ci_validate_and_cleanup_local_snapshot`, `ci_validate_weights_after_download`) that encapsulate CI validation and cleanup logic
- Simplify `weight_utils.py` by removing inline CI logic and using lazy imports gated by `is_in_ci()` checks
- `is_in_ci()` now appears only twice in `weight_utils.py`

**For users:** Simple download without validation overhead  
**For CI:** Automatic validation, cleanup, and retry on corruption

## Test plan

- [ ] CI tests pass
- [ ] Local testing as user (non-CI environment) to verify download works without overhead

alisonshao committed:
Refactor: separate CI-specific weight validation into dedicated module

Move CI-specific validation and cleanup logic from weight_utils.py into
a dedicated ci_weight_validation.py module. This improves code organization
by clearly separating user-facing functionality from CI-only behavior.

Changes:
- Rename weight_validation.py to ci_weight_validation.py
- Add ci_validate_and_cleanup_local_snapshot() wrapper function
- Add ci_validate_weights_after_download() wrapper function
- Simplify weight_utils.py by removing inline CI logic
- Use lazy imports for CI functions, gated by is_in_ci() checks
- is_in_ci() now appears only twice in weight_utils.py

For users: simple download without validation overhead
For CI: automatic validation, cleanup, and retry on corruption

Kangyan-Zhou commented on python/sglang/srt/model_loader/weight_utils.py:
```diff
@@ -584,6 +456,10 @@ def download_weights_from_hf(
 
         # Only perform validation and retry in CI to avoid overhead for regular users
         if is_in_ci():
```
Can we move this logic to the new file as well so that all the CI related logics are in the same file?

alisonshao committed:
Move CI download retry logic to ci_weight_validation.py

Address review feedback: consolidate all CI-related logic in the
dedicated ci_weight_validation.py module by creating a new
ci_download_with_validation_and_retry() wrapper function that
encapsulates the entire download-validate-retry flow.

This further simplifies weight_utils.py - now `is_in_ci()` gates
just a single function call instead of inline logic.

alisonshao committed:
Merge branch 'main' into refactor/weight-utils-ci-separation

alisonshao committed:
Merge branch 'main' into refactor/weight-utils-ci-separation

alisonshao committed:
Add HFRunner validation to detect and clean corrupted safetensors in CI

HFRunner uses transformers' from_pretrained() directly, bypassing SGLang's
weight validation. This caused cryptic 'EOF while parsing' errors when
corrupted safetensors files existed in the HF cache.

Added ci_validate_and_clean_hf_cache() that validates and cleans corrupted
files before HFRunner loads models, reusing existing validation logic.

Kangyan-Zhou committed:
Merge branch 'main' into refactor/weight-utils-ci-separation

alisonshao committed:
Address review comments: remove unused function, move import to top

- Remove unused find_local_hf_snapshot_dir function from weight_utils.py
- Move ci_validate_and_clean_hf_cache import to top of runners.py

alisonshao committed:
Merge branch 'main' into refactor/weight-utils-ci-separation

merrymercy commented on python/sglang/srt/model_loader/ci_weight_validation.py:
```diff
@@ -0,0 +1,629 @@
+    Args:
+        model_path: Model identifier (e.g., "meta-llama/Llama-2-7b")
+    """
+    from sglang.utils import is_in_ci
```
move to the top
